README.md

```diff
@@ -3,81 +3,135 @@
 
 Comparison:
-FastPolylines encode:    24561.8 i/s
-    Polylines encode:     3023.6 i/s - 8.12x  slower
+     FastPolylinesV2:   933750.7 i/s
+     FastPolylinesV1:    25715.1 i/s - 36.31x  slower
+           Polylines:     3254.3 i/s - 286.93x  slower
 
 
-** DECODING **
+———————————————————————————————  DECODING ————————————————————————————————
 
 Warming up --------------------------------------
-    Polylines decode   125.000  i/100ms
-FastPolylines decode     1.341k i/100ms
+           Polylines   140.000  i/100ms
+     FastPolylinesV1     1.602k i/100ms
+     FastPolylinesV2    36.432k i/100ms
 Calculating -------------------------------------
-    Polylines decode      1.284k (± 9.7%) i/s -      6.375k in   5.023536s
-FastPolylines decode     13.278k (±20.5%) i/s -     61.686k in   5.003858s
+           Polylines      1.401k (± 2.2%) i/s -      7.000k in   5.000321s
+     FastPolylinesV1     16.465k (± 3.7%) i/s -     83.304k in   5.067786s
+     FastPolylinesV2    396.100k (± 5.2%) i/s -      2.004M in   5.074500s
 
 Comparison:
-FastPolylines decode:    13278.0 i/s
-    Polylines decode:     1284.3 i/s - 10.34x  slower
+     FastPolylinesV2:   396100.0 i/s
+     FastPolylinesV1:    16464.6 i/s - 24.06x  slower
+           Polylines:     1400.6 i/s - 282.81x  slower
 ```
 
 ## Install
 
-```
+```bash
 gem install fast-polylines
 ```
 
-or in Bundler:
-```
-gem "fast-polylines"
+or in your `Gemfile`:
+```ruby
+gem "fast-polylines", "~> 2.0.0"
 ```
 
 ## Usage
 
-```
-require 'fast-polylines'
+```ruby
+require "fast-polylines"
 
-FastPolylines::Encoder.encode([[38.5, -120.2], [40.7, -120.95], [43.252, -126.453]])
+FastPolylines.encode([[38.5, -120.2], [40.7, -120.95], [43.252, -126.453]])
 # "_p~iF~ps|U_ulLnnqC_mqNvxq`@"
 
-FastPolylines::Decoder.decode("_p~iF~ps|U_ulLnnqC_mqNvxq`@")
+FastPolylines.decode("_p~iF~ps|U_ulLnnqC_mqNvxq`@")
 # [[38.5, -120.2], [40.7, -120.95], [43.252, -126.453]]
 ```
 
 ## Advanced usage
 
-*  With a different precision (default precision of `1e5`) :
+* With a different precision (default precision of `5` decimals):
+  ```ruby
+  FastPolylines.encode([[38.5, -120.2], [40.7, -120.95], [43.252, -126.453]], 6)
+  # "_izlhA~rlgdF_{geC~ywl@_kwzCn`{nI"
 
-```
-FastPolylines::Encoder.encode([[38.5, -120.2], [40.7, -120.95], [43.252, -126.453]], 1e6)
-# "_izlhA~rlgdF_{geC~ywl@_kwzCn`{nI"
+  FastPolylines.decode("_izlhA~rlgdF_{geC~ywl@_kwzCn`{nI", 6)
+  # [[38.5, -120.2], [40.7, -120.95], [43.252, -126.453]]
+  ```
+  The precision max is `14`.
 
-FastPolylines::Decoder.decode("_izlhA~rlgdF_{geC~ywl@_kwzCn`{nI", 1e6)
-# [[38.5, -120.2], [40.7, -120.95], [43.252, -126.453]]
+## Migrate from 1.0.0
+
+**TL;DR:**
+
+```ruby
+# before
+FastPolylines::Encoder.encode([[1.2, 1.2], [2.4, 2.4]], 1e6)
+# after
+FastPolylines.encode([[1.2, 1.2], [2.4, 2.4]], 6)
 ```
 
+The new version of `FastPolylines` doesn't support precision more than `1e10`,
+you should not consider using it anyway since [it is way too precise][xkcd].
```

Comment:
I think you are missing the link ;)

Reply:
Nop! All the links are at the bottom of the document to avoid cluttering and regroup them. And the rich diff shows me that it is fine as well :)